File: java/iroha_android/src/main/java/org/hyperledger/iroha/android/offline/OfflineJournalEntry.java
package org.hyperledger.iroha.android.offline;

import java.util.Arrays;

/** Immutable view over a pending offline journal entry. */
public final class OfflineJournalEntry {

  private final byte[] txId;
  private final byte[] payload;
  private final long recordedAtMs;
  private final byte[] hashChain;

  OfflineJournalEntry(
      final byte[] txId, final byte[] payload, final long recordedAtMs, final byte[] hashChain) {
    this.txId = Arrays.copyOf(txId, txId.length);
    this.payload = Arrays.copyOf(payload, payload.length);
    this.recordedAtMs = recordedAtMs;
    this.hashChain = Arrays.copyOf(hashChain, hashChain.length);
  }

  public byte[] txId() {
    return Arrays.copyOf(txId, txId.length);
  }

  public byte[] payload() {
    return Arrays.copyOf(payload, payload.length);
  }

  public long recordedAtMs() {
    return recordedAtMs;
  }

  public byte[] hashChain() {
    return Arrays.copyOf(hashChain, hashChain.length);
  }
}

File: kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/offline/OfflineJournalEntry.kt
package org.hyperledger.iroha.sdk.offline

/** Immutable view over a pending offline journal entry. */
class OfflineJournalEntry internal constructor(
    txId: ByteArray,
    payload: ByteArray,
    val recordedAtMs: Long,
    hashChain: ByteArray,
) {
    private val _txId: ByteArray = txId.copyOf()
    private val _payload: ByteArray = payload.copyOf()
    private val _hashChain: ByteArray = hashChain.copyOf()

    val txId: ByteArray get() = _txId.copyOf()
    val payload: ByteArray get() = _payload.copyOf()
    val hashChain: ByteArray get() = _hashChain.copyOf()
}
